include/swift/AST/PrintOptions.h

```diff
@@ -294,6 +275,9 @@ struct PrintOptions {
   /// Whether to print the extensions from conforming protocols.
   bool PrintExtensionFromConformingProtocols = false;
 
+  std::shared_ptr<ShouldPrintChecker> CurrentPrintabilityChecker =
```

Comment:
:-/ All right. `std::shared_ptr` just seems very heavyweight to me, but I guess PrintOptions don't get copied around a lot and customizing this behavior doesn't happen very often anyway.